airgun/views/computeresource.py

```diff
@@ -199,34 +218,207 @@ class compute_profiles(SatTab):
             }
         )
 
+    class images(SatTab, SearchableViewMixin):
+        TAB_NAME = 'Images'
+        table = ComputeResourceTabTable(
+            './/table',
+            column_widgets={
+                'Actions': ActionsDropdown("./div[contains(@class, 'btn-group')]"),
+            }
+        )
+
+
+class RemovableWidgetsItemsListView(View):
+    """A host for widgets list. Items that can added and removed, mainly used in profile for
+    network interfaces and storage.
+
+    Usage::
+
+        @View.nested
+        class resources(RemovableWidgetsItemsListView):
+            ROOT = "//fieldset[@id='storage_volumes']"
+            ITEMS = "./div/div[contains(@class, 'removable-item')]"
+            ITEM_WIDGET_CLASS = ComputeResourceRHVProfileStorageItem
+    """
+    ITEMS = "./div[contains(@class, 'removable-item')]"
+    ITEM_WIDGET_CLASS = None
+    ITEM_REMOVE_BUTTON_ATTR = 'remove_button'
+    add_item_button = Text(".//a[contains(@class, 'add_nested_fields')]")
+
+    def _get_item_locator(self, index):
+        """Return the item locator located at index position"""
+        return '{0}[{1}]'.format(self.ITEMS, index + 1)
+
+    def get_item_at_index(self, index):
+        """Return the item widget instance at index"""
+        return self.ITEM_WIDGET_CLASS(self, self._get_item_locator(index))
+
+    def add_item(self):
+        """Add an item by pressing the add_item button and return the item instance"""
+        self.add_item_button.click()
+        return self.get_item_at_index(self.items_length - 1)
+
+    def remove_item(self, item):
+        """Remove item widget by clicking on it's remove button"""
+        if self.ITEM_REMOVE_BUTTON_ATTR:
+            getattr(item, self.ITEM_REMOVE_BUTTON_ATTR).click()
+
+    def remove_item_at_index(self, index):
+        """Remove item at index"""
+        self.remove_item(self.get_item_at_index(index))
+
+    @property
+    def items_length(self):
+        # Returns the items length
+        return len(self.browser.elements(self.ITEMS, parent=self))
+
+    @property
+    def items(self):
+        """Return all the items widget instances"""
+        return [self.get_item_at_index(index) for index in range(self.items_length)]
+
+    def clear(self):
+        """Remove all items if item remove button att defined."""
+        if self.ITEM_REMOVE_BUTTON_ATTR:
+            for item in reversed(self.items):
+                self.remove_item(item)
+
+    def read(self):
+        """Read all items"""
+        return [item.read() for item in self.items]
+
+    def fill(self, values):
+        """Fill all items.
+        :param values: A list of values to fill the item widgets with.
+        """
+        if not values:
+            values = []
```

Comment:
why not `return False`?

Reply:
if we need a way to clear the form, so fill(None) will clear the form,  return False, will do nothing here ,otherwise we can return False

Reply:
If that is a thing you need then it is fine, but maybe you have to call `clear` when you want to clear? as that will make more sense

Reply:
This is not that evident (This is not a first class view but embedded in an other view it's even not an entity view but a sub entity view and you have not direct access to that clear function) as the whole view is considered as a widget, as input box when you pass none it will be cleared first. if we had clear that list implicitly , we have to create two function in entity clear_rhv_image_interface_networks and clear_rhv_image_storage and this functions will be relevant only to RHV. It's more simple to clear them by passing a None or empty list.